Assets/Plugins/Mediabox/Source/GameKit/GameManager/PauseHandler.cs

```diff
@@ -2,30 +2,58 @@
 
 namespace Mediabox.GameKit.GameManager {
     public class PauseHandler {
-        float timeScaleBeforePause;
-        float volumeBeforePause;
-        bool isPaused;
+        float timeScaleBeforePause = 1;
+        float volumeBeforePause = 1;
+        bool isPaused = false;
+
+        public delegate void OnPause(bool pausing);
+        public event OnPause onPause;
+        
         
         public void Pause() {
-            if (!this.isPaused) {
-                this.timeScaleBeforePause = Time.timeScale;
-                this.volumeBeforePause = AudioListener.volume;
-                this.isPaused = true;
-            } else {
+            
+            
+            if (this.isPaused){
                 Debug.LogWarning("Pause() has been called multiple times. To avoid unintended behavior, it is recommended to ensure calling Unpause() before Pausing again.");
             }
-            Time.timeScale = 0f;
+            
+            this.isPaused = true;
+            this.volumeBeforePause = AudioListener.volume;
```

Comment:
I figured this would allow mediabox to reset the game back to a playable state if all else failed for the user. For example, something happens when pausing that messes up the game. Instead of being locked in a time.timescale = 0 state. The user can use the wonderz button (blue arrow) to exit the game full, pick the game again (or a new game), now the values for timeBeforePause have been hard reset. They can enter the game with the timescale corrected, and they can continue on playing without having to force quit the app.

Reply:
Basically if they get to this point, we've already messed up, and we get the error flag while testing. But for the end user the product is still "serviceable" even while broken.